mmv1/third_party/tgc_next/pkg/cai2hcl/converters/utils/utils.go

```diff
@@ -22,6 +22,44 @@ func ParseFieldValue(url string, name string) string {
 	return ""
 }
 
+/*
+	ParseUrlParamValuesFromAssetName uses CaiAssetNameTemplate to parse hclData from assetName
+
+template: //bigquery.googleapis.com/projects/{{project}}/datasets/{{dataset_id}}
+assetName: //bigquery.googleapis.com/projects/my-project/datasets/my-dataset
+hclData: [project:my-project dataset_id:my-dataset]
+*/
+func ParseUrlParamValuesFromAssetName(assetName string, template string, hclData map[string]any) {
+	fragments := strings.Split(template, "/")
+	if len(fragments) < 2 {
+		// We need a field and a prefix.
+		return
+	}
+	fields := make(map[string]string) // keys are prefixes in URI, values are names of fields
+	for ix, item := range fragments[1:] {
+		if trimmed, ok := strings.CutPrefix(item, "{{"); ok {
+			if trimmed, ok = strings.CutSuffix(trimmed, "}}"); ok {
+				fields[fragments[ix]] = trimmed // ix is relative to the subslice
+			}
+		}
+	}
+	fragments = strings.Split(assetName, "/")
+	for ix, item := range fragments[:len(fragments)-1] {
+		if fieldName, ok := fields[item]; ok {
+			hclData[fieldName] = fragments[ix+1]
+		}
+	}
+}
+
+// DeleteOutputFields deletes the given outputFields from the given hclData.
+func DeleteOutputFields(outputFields map[string]struct{}, hclData map[string]any) {
+	for field := range hclData {
+		if _, isOutput := outputFields[field]; isOutput {
```

Comment:
What about moving this if statement to line 48?

Reply:
You mean like this?